artiq/coredevice/comm_kernel.py

```diff
@@ -436,13 +436,15 @@ def check(cond, expected):
                   lambda: "bool")
             self._write_bool(value)
         elif tag == "i":
+            # notice: let b be the number of bits, check [-2^(b-1), 2^(b-1))
+            # equivalent: check [0, 2^b)
```

Comment:
Not sure what this comment means or if it adds anything to the code.

Reply:
that's the explanation for the code change

Reply:
Put it in the PR description (if anywhere), not comment. It's quite obvious already what is going on, that was just a silly mistake...

Reply:
Okay. This off by one mistake is very common by the way.